browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,472 @@
+        base::UTF8ToUTF16(*BraveCosmeticResourcesTabHelper::observing_script_),
+        base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+    }
+    // Working on css rules, we do that on a main frame only
+    if (!do_non_scriplets)
+      return;
+    CSSRulesRoutine(url, resources_dict, frame_id);
+  }
+}
+
+void BraveCosmeticResourcesTabHelper::CSSRulesRoutine(
+    const std::string& url_string, base::DictionaryValue* resources_dict,
+    content::GlobalFrameRoutingId frame_id) {
+  const GURL url(url_string);
+  if (url.is_empty() || !url.is_valid() || IsVettedSearchEngine(url.host())) {
+    return;
+  }
+
+  base::ListValue* cf_exceptions_list;
+  if (resources_dict->GetList("exceptions", &cf_exceptions_list)) {
+    for (size_t i = 0; i < cf_exceptions_list->GetSize(); i++) {
+      exceptions_.push_back(cf_exceptions_list->GetList()[i].GetString());
+    }
+  }
+  base::ListValue* hide_selectors_list;
+  if (resources_dict->GetList("hide_selectors", &hide_selectors_list)) {
+    std::string json_selectors;
+    if (!base::JSONWriter::Write(*hide_selectors_list, &json_selectors) ||
+        json_selectors.empty()) {
+      json_selectors = "[]";
+    }
+    // Building a script for stylesheet modifications
+    std::string new_selectors_script =
+        base::StringPrintf(k_selectors_inject_script, json_selectors.c_str());
+    if (hide_selectors_list->GetSize() != 0) {
+      auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+      if (!frame_host)
+        return;
+      frame_host->ExecuteJavaScriptInIsolatedWorld(
+          base::UTF8ToUTF16(new_selectors_script),
+          base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+    }
+  }
+
+  base::DictionaryValue* style_selectors_dictionary = nullptr;
+  if (resources_dict->GetDictionary("style_selectors",
+      &style_selectors_dictionary)) {
+    std::string json_selectors;
+    if (!base::JSONWriter::Write(
+        *style_selectors_dictionary, &json_selectors) ||
+            json_selectors.empty()) {
+      json_selectors = "[]";
+    }
+    std::string new_selectors_script =
+        base::StringPrintf(k_style_selectors_inject_script,
+            json_selectors.c_str());
+    if (!json_selectors.empty()) {
+      auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+      if (!frame_host)
+        return;
+      frame_host->ExecuteJavaScriptInIsolatedWorld(
+          base::UTF8ToUTF16(new_selectors_script),
+          base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+    }
+  }
+
+  if (!enabled_1st_party_cf_filtering_) {
+    auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+    if (!frame_host)
+      return;
+    frame_host->ExecuteJavaScriptInIsolatedWorld(
+        base::UTF8ToUTF16(
+            *BraveCosmeticResourcesTabHelper::observing_script_),
+        base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+  }
+}
+
+void BraveCosmeticResourcesTabHelper::GetHiddenClassIdSelectorsOnUI(
```

Comment:
This rather not `Get...` but `Inject.. ` ? Or `Apply`..

Reply:
That function is related to the function that pulls rules from IO thread and pass the result to UI thread, I think that keep there names similar is better:
`base::BindOnce(&GetHiddenClassIdSelectorsOnTaskRunner, classes, ids,
              exceptions_),
          base::BindOnce(&BraveCosmeticResourcesTabHelper::
              GetHiddenClassIdSelectorsOnUI, AsWeakPtr(),
              content::GlobalFrameRoutingId(
                  render_frame_host->GetProcess()->GetID(),
                  render_frame_host->GetRoutingID()),
              web_contents()->GetLastCommittedURL()));`